src/geo/ui/legends/base/legend-view-base.js

```diff
@@ -21,41 +19,35 @@ var LegendViewBase = Backbone.View.extend({
 
     this._loadImages();
 
-    if (this.model.isVisible()) {
-      this.$el.show();
-    } else {
-      this.$el.hide();
-    }
+    this.model.isVisible()
+      ? this.$el.show()
+      : this.$el.hide();
 
     this._toggleLoadingClass();
 
     return this;
   },
 
   _generateHTML: function () {
-    var html = [];
-    if (this.model.isSuccess()) {
-      if (this.model.isAvailable()) {
-        html.push(this._getLegendHTML());
-      } else {
-        html.push(this._getNoDataAvailableHTML());
-        html.push(this._getPlaceholderHTML());
-      }
-    } else if (this.model.isError()) {
-      html.push(this._getErrorHeaderHTML());
-      html.push(this._getPlaceholderHTML());
-    } else if (this.model.isLoading()) {
-      html.push(this._getPlaceholderHTML());
+    if (this.model.isSuccess() && this.model.isAvailable()) {
+      return this._getLegendHTML();
     }
-    return html.join('\n');
+
+    var placeholder = this._getPlaceholderHTML();
+
+    var error = this.model.isError()
```

Comment:
no loading state?

Reply:
Well, that as a fail 😓  let me change it